.github/workflows/reusable-agent-build-container-images.yml

```diff
@@ -74,33 +85,52 @@ jobs:
 
   # Build cacheable requirements of the images.
   build_image_requirements:
-    name: requirements-${{ matrix.builder_name }}-${{ matrix.architecture }}
-    runs-on: ubuntu-22.04
+    name: requirements-${{ inputs.builder_name }}-${{ matrix.architecture }}
+    runs-on: ${{ vars.RUNNER }}
     strategy:
       matrix:
-        builder_name: ["ubuntu", "alpine"]
-        architecture: ["x86_64", "aarch64", "armv7"]
-
+        architecture: ${{ fromJson(inputs.architectures) }}
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4
 
       - name: install python and requirements
         uses: ./.github/actions/install_python_and_requirements
         with:
           python_version: ${{ inputs.python_version }}
 
+      - name: Login to DockerHub
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3
+        with:
+          username: ${{ secrets.PUBLISH_REGISTRY_USERNAME }}
+          password: ${{ secrets.PUBLISH_REGISTRY_PASSWORD }}
+
+      - name: Login to an internal build registry
```

Comment:
Should we only run this step in case some variable (e.g. runner) indicates we are building it for internal registry / using on prem runner?

Reply:
EDIT: I see build is failing due to missing username and password, so we likely indeed only want to run this step in case `INTERNAL_BUILD_REGISTRY_USERNAME` and `INTERNAL_BUILD_REGISTRY_PASSWORD` is defined / not empty.

Reply:
Yeah, I wanted to let if fail and continue in case there are no credentials for it. But it will be cleaner to do it the way you suggested.